lib/dev-middleware.js

```diff
@@ -92,6 +92,20 @@ module.exports = function (app, config, options) {
                         }
                     }
 
+                    if (options.virtualWrite) {
+                        let vw_root = path.normalize(path.resolve(options.virtualWrite))
+                            .replace(process.cwd(), '')
```

Comment:
I'm not a fan of this replace cwd strategy. Seems safe on Windows, but POSIX paths can very credibly contain the whole `cwd` path somewhere in their middle.

`/projects/bar/docs/src/projects/foo` would break with cwd `/projects/foo`... Even if I can't think of a real case example in the current situation, this seems like a dangerous habit.

Reply:
In theory even on POSIX this should work just fine. When you call "resolve", it will always give an absolute path, which will always start with the current working directory. Replace only replaces the first match it finds, so even if the cwd is replicated in the middle somewhere, it won't match, so it should always replace the start of the resolved path with empty. 

Turns out though there's ```path.join``` which I wasn't aware of that does exactly what I need! Will switch to that! :)

Reply:
Do you mean `path.relative`?

> "resolve", it will always give an absolute path, which will always start with the current working directory

My point is a Windows absolute path `c:\whatever` can probably only appear at the beginning of an absolute path (i.e. you can get `c:\somethingelse\c:\whatever`) but, with POSIX, `/absolute/path` can appear in the middle of `/another/absolute/path`.

From what I can tell, the problematic case cannot be met in the precise situation we're talking about, but I think this is an unsafe pattern and should be best avoided always.